### DISCUSSION TITLE
need help in syntax

### DISCUSSION THREAD

**Discussion by A123-bansal123** [firedrakeproject/firedrake#2969] [Firedrake support] [answered]
I write a fenics syntax and I want to write same syntax in firedrake but able to write. PLease help.
```
from ufl import JacobianInverse, indices
K = JacobianInverse(M)
i, j, k = indices(3)
G = as_tensor(K[k, i]*K[k, j], (i, j))
g = as_vector( [sum(K[k, i] for k in range(M.geometry().dim())) for i in range(M.geometry().dim())])
```
error 
```
 File "vms.py", line 38, in <module>
    g = as_vector( [sum(K[k, i] for k in range(M.geometry().dim())) for i in range(M.geometry().dim())])
  File "/home/aparna/software/firedrake/src/firedrake/firedrake/mesh.py", line 2153, in __getattr__
    val = getattr(self._topology, name)
AttributeError: 'MeshTopology' object has no attribute 'geometry'

```

**Comment by A123-bansal123** [comment]


**Reply by dham** [reply]
Can I just gently remind you not to assume that others work your working hours. The core Firedrake team are in the UK. Your initial request came in at 20:07 in our timezone and you followed up asking for attention at 05:59 this morning. We were all asleep for most of the intervening period, and most of us won't be working at 8pm. It's generally polite to wait at least 48 hours before following up because of timezones and the fact that people may be otherwise busy.

**Reply by A123-bansal123** [reply]
ok

**Comment by connorjward** [comment, marked as answer]
I think you want `M.geometric_dimension()` or `M.ufl_cell().geometric_dimension()`.
